src/SignalR/server/StackExchangeRedis/src/RedisHubLifetimeManager.cs

```diff
@@ -388,6 +395,78 @@ public void Dispose()
         _ackHandler.Dispose();
     }
 
+    /// <inheritdoc/>
+    public override async Task<T> InvokeConnectionAsync<T>(string connectionId, string methodName, object?[] args, CancellationToken cancellationToken = default)
+    {
+        // send thing
+        if (connectionId == null)
+        {
+            throw new ArgumentNullException(nameof(connectionId));
+        }
+
+        var connection = _connections[connectionId];
+
+        // Needs to be unique across servers, easiest way to do that is prefix with connection ID.
+        var invocationId = $"{connectionId}{Interlocked.Increment(ref _lastInvocationId)}";
```

Comment:
This seems fine as long as it's not being sent all the way to the client.

Reply:
So this is being sent all the way to the client and our idea of storing a tuple of (connectionId, invocationId) won't work.

The invocationId needs to be the main source of truth for identifying a specific invoke as it is used by the HubProtocols when querying for the expected type for deserialization. If we stored as a (connectionId, invocationId) then we won't have the connectionId when asking the dictionary for the type. We could in theory iterate over every item and find the invocationId, but then it's possible to have duplicates for IDs coming from different servers.

TLDR; The invocationId needs to be unique across all servers. There are 3 ways to do this, use the connection ID as part of the invocationId (what I did), use Redis to store a key and have all servers accessing and incrementing it (please no), or generate another unique key like we do for connection IDs to represent the invocationId.

Reply:
I still don't see why any of this would mean the connection id needs to be transmitted back and forth from the client. I could imagine some sort of low-level transport filter that added the client's connection id to the invocation id on the way to the server and strips it on the way out. This wouldn't have to change the internal logic on the server at all.

Reply:
The problem is HubProtocol only has the invocationID when asking for the type. If servers don't generate a unique ID across all servers then there could be multiple of the same IDs.

Reply:
That's an interesting idea! I'll think about it for preview5